sdk/spring/spring-cloud-azure-starter-monitor/src/main/java/com/azure/spring/cloud/autoconfigure/monitor/AzureSpringMonitorAutoConfiguration.java

```diff
@@ -81,42 +59,37 @@ private static boolean applicationInsightsAgentIsAttached() {
-                        + " You can remove this message by adding the otel.sdk.disabled=true property.");
-                    autoConfigurationCustomizer.addPropertiesCustomizer(OTEL_DISABLE_CONFIG);
-                    return;
-                }
-
-                if (connectionString == null || connectionString.isEmpty()) {
-                    LOG.warn("Unable to find the Application Insights connection string. The telemetry data won't be sent to Azure.");
-                    // If the user does not provide a connection, we disable the export and leave the instrumentation enabled to spot potential failures from
-                    // the instrumentation, in the customer automatic tests for example.
-                    autoConfigurationCustomizer.addPropertiesCustomizer(NO_EXPORT_CONFIG);
-                    return;
-                }
-
-                if (!connectionString.contains("InstrumentationKey=")) {
-                    throw new WrongConnectionStringException();
-                }
-
-                AutoConfiguredOpenTelemetrySdkBuilder sdkBuilder = (AutoConfiguredOpenTelemetrySdkBuilder) autoConfigurationCustomizer;
-
-                HttpPipeline providedHttpPipeline = httpPipeline.getIfAvailable();
-                if (providedHttpPipeline != null) {
-                    AzureMonitorExporterBuilder azureMonitorExporterBuilder = new AzureMonitorExporterBuilder().connectionString(connectionString).pipeline(providedHttpPipeline);
-                    AzureMonitor.customize(sdkBuilder, azureMonitorExporterBuilder);
-                } else {
-                    AzureMonitor.customize(sdkBuilder, connectionString);
-                }
+    AutoConfigurationCustomizerProvider autoConfigurationCustomizerProvider(@Value("${applicationinsights.connection.string:#{null}}") String connectionString, ObjectProvider<AzureMonitorExporterBuilder> azureMonitorExporterBuilder) {
+
+        if (!isNativeRuntimeExecution() && applicationInsightsAgentIsAttached()) {
+            LOG.warn("The spring-cloud-azure-starter-monitor Spring starter is disabled because the Application Insights Java agent is enabled."
+                + " You can remove this message by using the otel.sdk.disabled=true property.");
+            return DISABLE_OTEL_CUSTOMER_PROVIDER;
+        }
+
+        AzureMonitorExporterBuilder providedAzureMonitorExporterBuilder = azureMonitorExporterBuilder.getIfAvailable();
+        if (providedAzureMonitorExporterBuilder != null) {
+            if (System.getenv("APPLICATIONINSIGHTS_CONNECTION_STRING") == null && System.getProperty("applicationinsights.connection.string") == null && (connectionString != null && !connectionString.isEmpty())) {
+                LOG.warn("You have created an AzureMonitorExporterBuilder bean and set the applicationinsights.connection.string property in a .properties or .yml file."
+                    + " This property is ignored.");
             }
```

Comment:
oh, I understand what this code is doing now, thanks

what about instead?

```suggestion
            if (connectionString != null && !connectionString.isEmpty()) {
                providedAzureMonitorExporterBuilder.connectionString(connectionString);
            }
```

Reply:
No, because the user could programmatically configure the connection string and the code will systematically override the value.